qiskit/aqua/algorithms/adaptive/vqe/vqe.py

```diff
@@ -219,28 +216,6 @@ def construct_circuit(self, parameter, backend=None, use_simulator_operator_mode
                                                               input_circuit, backend, use_simulator_operator_mode)
         return circuit
 
-    def _solve(self):
-        opt_params, opt_val = self.find_minimum_eigenvalue()
-        self._ret['eigvals'] = np.asarray([opt_val])
-        self._ret['opt_params'] = opt_params
-        qc = self._var_form.construct_circuit(self._ret['opt_params'])
-        if self._quantum_instance.is_statevector:
-            ret = self._quantum_instance.execute(qc)
-            self._ret['eigvecs'] = np.asarray([ret.get_statevector(qc)])
-        else:
-            c = ClassicalRegister(self._operator.num_qubits, name='c')
-            q = find_regs_by_name(qc, 'q')
-            qc.add_register(c)
-            qc.barrier(q)
-            qc.measure(q, c)
-            ret = self._quantum_instance.execute(qc)
-            self._ret['eigvecs'] = np.asarray([ret.get_counts(qc)])
-
-    def _get_ground_state_energy(self):
-        if 'eigvals' not in self._ret:
-            self._solve()
-        self._ret['energy'] = self._ret['eigvals'][0]
-
     def _eval_aux_ops(self, threshold=1e-12):
         if 'opt_params' not in self._ret:
             self._get_ground_state_energy()
```

Comment:
in case that this branch is executed, since _get_ground_state_energy is undefined, the execution of _get_ground_state_energy will lead to the following error:
AttributeError: 'VQE' object has no attribute '_get_ground_state_energy'
This looks like a bug or code smell.

Reply:
instead of calling the non-existent method _get_ground_state_energy, you can simply throw an exception.
Or if you are sure the if-branch is always false (now and in future), you can remove the if-branch.

Reply:
Was an error, fixed.